docs/how-to-use-aws-sdk-java-encryption.md

```diff
@@ -0,0 +1,143 @@
+# How to use AWS SDK for Java for encryption with Minio Server [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
+
+`aws-sdk` for Java is the official AWS SDK for Java. In this recipe we will learn how to use `aws-sdk` for Java for encryption on Minio server using both Symmetric and Asymmetric approach.
+
+Encryption allows additional security layer for sensitive user data (images, audio clips, etc.) stored on Minio server.
+
+## Prerequisites
+
+Install `aws-sdk` for Java from the official AWS SDK docs [here](http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/welcome.html).
+
+## Symmetric Encryption
+
+Symmetric Encryption uses single key for both encryption and decryption. Following are the steps involved for Symmetric Encryption using AES :
+
+### 1. Generate AES key
+
+ Use [Java KeyGenrator](https://docs.oracle.com/javase/7/docs/api/javax/crypto/KeyGenerator.html) class for generating 256 bit AES key.
+
+```java
+    //Generate symmetric 256 bit AES key.
+    KeyGenerator symKeyGenerator = KeyGenerator.getInstance("AES");
+    symKeyGenerator.init(256);
+    SecretKey symKey = symKeyGenerator.generateKey();
+```
+
+### 2. Create AWS S3 encryption client using generated key.
+
+```java
+    EncryptionMaterials encryptionMaterials = new EncryptionMaterials(
+      mySymmetricKey);
+
+    // Add Minio server accessKey and secretKey  
+    AWSCredentials credentials = new BasicAWSCredentials(
+      "USWUXHGYZQYFYFFIT3RE", "MOJRH0mkL1IPauahWITSVvyDrQbEEIwljvmxdq03");
+
+    // Create encryption client with Minio server as endpoint  
+    AmazonS3EncryptionClient encryptionClient = new AmazonS3EncryptionClient(
+      credentials, new StaticEncryptionMaterialsProvider(
+      encryptionMaterials));
+    Region usEast1 = Region.getRegion(Regions.US_EAST_1);
+    encryptionClient.setRegion(usEast1);
+    encryptionClient.setEndpoint("http://localhost:9000");
```

Comment:
We can use examples to have <https://play.minio.io:9000> here instead. So that users can try this in drop in replacement.